src/freertos_drivers/common/SPI.hxx

```diff
@@ -54,6 +55,49 @@ public:
     /** Function point for the chip select assert and deassert methods */
     typedef void (*ChipSelectMethod)();
 
+    /** Method to transmit/receive the data.  Chip select will be asserted at
+     * start of the transfer and deasserted at the end of the transfer.
+     * @param self pointer to a SPI object instance
+     * @param msg message(s) to transact.
+     * @return bytes transfered upon success, -errno upon failure
+     */
+    __attribute__((optimize("-O3")))
+    static int transfer(SPI *self,struct spi_ioc_transfer *msg)
```

Comment:
let's not make this static but a regular instance function.
if there is a difference in assertion behavior, let's include that int he function name, like transfer_with_cs_assert.

Reply:
Fixed.